libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -401,12 +402,29 @@ public MapboxMap getMapboxMap() {
     return map;
   }
 
+  public void setInstructionViewCallback() {
+    if (instructionView != null) {
+      instructionView.setInstructionListListener(new InstructionListListener() {
+        @Override
+        public void onInstructionListShown(boolean shown) {
+          if (shown) {
```

Comment:
This business logic can be extracted into `NavigationPresenter` so we can call `navigationPresenter.onInstructionListVisibilityChanged(isShowing)`.  The show / hide recenter button are already view methods that the presenter has access to

Reply:
@danesfeder hmm maybe the dispatcher should reside in the presenter